File: PolicyPlus/DetailPolicy.vb
Public Class DetailPolicy
    Dim SelectedPolicy As PolicyPlusPolicy
    Public Sub PresentDialog(Policy As PolicyPlusPolicy)
        SelectedPolicy = Policy
        NameTextbox.Text = Policy.DisplayName
        IdTextbox.Text = Policy.UniqueID
        DefinedTextbox.Text = Policy.RawPolicy.DefinedIn.SourceFile
        DisplayCodeTextbox.Text = Policy.RawPolicy.DisplayCode
        InfoCodeTextbox.Text = Policy.RawPolicy.ExplainCode
        PresentCodeTextbox.Text = Policy.RawPolicy.PresentationID
        Select Case Policy.RawPolicy.Section
            Case AdmxPolicySection.Both
                SectionTextbox.Text = "User or computer"
            Case AdmxPolicySection.Machine
                SectionTextbox.Text = "Computer"
            Case AdmxPolicySection.User
                SectionTextbox.Text = "User"
        End Select
        SupportButton.Enabled = Policy.SupportedOn IsNot Nothing
        If Policy.SupportedOn IsNot Nothing Then
            SupportTextbox.Text = Policy.SupportedOn.DisplayName
        ElseIf Policy.RawPolicy.SupportedCode <> "" Then
            SupportTextbox.Text = "<missing: " & Policy.RawPolicy.SupportedCode & ">"
        Else
            SupportTextbox.Text = ""
        End If
        CategoryButton.Enabled = Policy.Category IsNot Nothing
        If Policy.Category IsNot Nothing Then
            CategoryTextbox.Text = Policy.Category.DisplayName
        ElseIf Policy.RawPolicy.CategoryID <> "" Then
            CategoryTextbox.Text = "<orphaned from " & Policy.RawPolicy.CategoryID & ">"
        Else
            CategoryTextbox.Text = "<uncategorized>"
        End If
        ShowDialog()
    End Sub
    Private Sub SupportButton_Click(sender As Object, e As EventArgs) Handles SupportButton.Click
        DetailSupport.PresentDialog(SelectedPolicy.SupportedOn)
    End Sub
    Private Sub CategoryButton_Click(sender As Object, e As EventArgs) Handles CategoryButton.Click
        DetailCategory.PresentDialog(SelectedPolicy.Category)
    End Sub
End Class

File: PolicyPlus/DetailPolicy.cs
using System;

namespace PolicyPlus
{
    public partial class DetailPolicy
    {
        private PolicyPlusPolicy SelectedPolicy;

        public DetailPolicy()
        {
            InitializeComponent();
        }
        public void PresentDialog(PolicyPlusPolicy Policy)
        {
            SelectedPolicy = Policy;
            NameTextbox.Text = Policy.DisplayName;
            IdTextbox.Text = Policy.UniqueID;
            DefinedTextbox.Text = Policy.RawPolicy.DefinedIn.SourceFile;
            DisplayCodeTextbox.Text = Policy.RawPolicy.DisplayCode;
            InfoCodeTextbox.Text = Policy.RawPolicy.ExplainCode;
            PresentCodeTextbox.Text = Policy.RawPolicy.PresentationID;
            switch (Policy.RawPolicy.Section)
            {
                case AdmxPolicySection.Both:
                    {
                        SectionTextbox.Text = "User or computer";
                        break;
                    }
                case AdmxPolicySection.Machine:
                    {
                        SectionTextbox.Text = "Computer";
                        break;
                    }
                case AdmxPolicySection.User:
                    {
                        SectionTextbox.Text = "User";
                        break;
                    }
            }
            SupportButton.Enabled = Policy.SupportedOn is not null;
            if (Policy.SupportedOn is not null)
            {
                SupportTextbox.Text = Policy.SupportedOn.DisplayName;
            }
            else if (!string.IsNullOrEmpty(Policy.RawPolicy.SupportedCode))
            {
                SupportTextbox.Text = "<missing: " + Policy.RawPolicy.SupportedCode + ">";
            }
            else
            {
                SupportTextbox.Text = "";
            }
            CategoryButton.Enabled = Policy.Category is not null;
            if (Policy.Category is not null)
            {
                CategoryTextbox.Text = Policy.Category.DisplayName;
            }
            else if (!string.IsNullOrEmpty(Policy.RawPolicy.CategoryID))
            {
                CategoryTextbox.Text = "<orphaned from " + Policy.RawPolicy.CategoryID + ">";
            }
            else
            {
                CategoryTextbox.Text = "<uncategorized>";
            }
            ShowDialog();
        }
        private void SupportButton_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailSupport.PresentDialog(SelectedPolicy.SupportedOn);
        }
        private void CategoryButton_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailCategory.PresentDialog(SelectedPolicy.Category);
        }
    }
}